File: 1002-find-common-characters/1002-find-common-characters.cpp
class Solution {
public:
    vector<string> commonChars(vector<string>& words) {

        vector<int> common(26, 0);

        for(char ch : words[0]) {
            common[ch - 'a']++;
        }

        for(int i = 1; i < words.size(); i++) {

            vector<int> curr(26, 0);

            for(char ch : words[i]) {
                curr[ch - 'a']++;
            }

            for(int j = 0; j < 26; j++) {
                common[j] = min(common[j], curr[j]);
            }
        }

        vector<string> ans;

        for(int i = 0; i < 26; i++) {
            while(common[i]--) {
                ans.push_back(string(1, 'a' + i));
            }
        }

        return ans;
    }
};

File: 1002-find-common-characters/1002-find-common-characters.py
class Solution:
    def commonChars(self, words: List[str]) -> List[str]:
        mp={}
        for x in words[0]:
            mp[x] = mp.get(x,0)+1
        for i in range(len(words)):
            mp1={}
            for ch in words[i]:
                mp1[ch] = mp1.get(ch, 0) + 1
            
            for ch in mp:
                mp[ch] = min(mp[ch], mp1.get(ch, 0))

        ans = []

        for ch in mp:
            for j in range(mp[ch]):
                ans.append(ch)
        return list(ans)
            